src/runtime/naming/name.cpp

```diff
@@ -278,53 +296,68 @@ namespace hpx { namespace naming
 
                     util::unlock_guard<scoped_lock> ul(l);
                     boost::int64_t new_credit = (HPX_GLOBALCREDIT_INITIAL - 1) * 2;
-                    agas::incref(new_gid, new_credit);
+                    new_gid = gid;
+                    HPX_ASSERT(new_gid != invalid_gid);
+                    return agas::incref_async(new_gid, new_credit).then(
+                        [&gid](future<boost::int64_t>&&) mutable -> gid_type
+                        {
+                            typedef gid_type::mutex_type::scoped_lock scoped_lock;
+                            scoped_lock ll(gid.get_mutex());
+                            gid_type new_gid = gid;
```

Comment:
Am 20.08.2015 2:42 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In src/runtime/naming/name.cpp:
>
> > @@ -278,53 +296,68 @@ namespace hpx { namespace naming
> >
> >                      util::unlock_guard<scoped_lock> ul(l);
> >                      boost::int64_t new_credit =
(HPX_GLOBALCREDIT_INITIAL - 1) * 2;
> > -                    agas::incref(new_gid, new_credit);
> > +                    new_gid = gid;
> > +                    HPX_ASSERT(new_gid != invalid_gid);
> > +                    return agas::incref_async(new_gid,
new_credit).then(
> > +                        [&gid](future<boost::int64_t>&&) mutable ->
gid_type
> > +                        {
> > +                            typedef gid_type::mutex_type::scoped_lock
scoped_lock;
> > +                            scoped_lock ll(gid.get_mutex());
> > +                            gid_type new_gid = gid;
>
> Is it guaranteed that gid is still alive when the lambda is executed?

Yes, the gid is a reference to the one that is supposed to be serialized.

>
> —
> Reply to this email directly or view it on GitHub.